app/controllers/main_controller.rb

```diff
@@ -8,4 +8,30 @@
 
 # Your controller actions go below this line.
 # -----------------------------------------------------------------------------
+# require_relative "../lib/dictionary"
+# require_relative "../lib/string_segmenter"
 
+MyApp.get "/" do
+  erb :"homepage"
+end
+
+MyApp.get "/admin" do
+  erb :"/a/admin"
+end
+
+
+
+
+MyApp.get "/segmenter" do
+  
+  @segment = StringSegmenter.new(params[:string])
```

Comment:
I would expect you to be running `@segment.run_program` and `x = @segment.final_words` somewhere in here.